File: tools/package-release.mjs
import { execSync } from 'node:child_process';
import { statSync, existsSync, unlinkSync } from 'node:fs';
import { resolve, dirname } from 'node:path';
import { fileURLToPath } from 'node:url';

const ROOT = dirname(dirname(fileURLToPath(import.meta.url)));
const distDir = resolve(ROOT, 'dist', 'POE-Ninja-three-language-v0.2.13');
const zipPath = resolve(ROOT, 'dist', 'POE-Ninja-three-language-v0.2.14.zip');

if (existsSync(zipPath)) {
  unlinkSync(zipPath);
}

console.log('📦 正在生成最终可分发生产安装包 (ZIP)...');
console.log('源目录:', distDir);
console.log('目标包:', zipPath);

const psCommand = `powershell -NoProfile -Command "Compress-Archive -Path '${distDir}\\*' -DestinationPath '${zipPath}' -Force"`;
execSync(psCommand, { stdio: 'inherit' });

const stat = statSync(zipPath);
console.log('\n====================================================');
console.log('🎉 最终可分发压缩包已成功生成！');
console.log(`📍 文件绝对路径: ${zipPath}`);
console.log(`📦 安装包最终体积: ${(stat.size / 1024 / 1024).toFixed(2)} MB (${(stat.size / 1024).toFixed(1)} KB)`);
console.log('====================================================');

File: tools/package-release.py
import os
import sys
import zipfile

try:
    sys.stdout.reconfigure(encoding="utf-8")
except Exception:
    pass

root = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
dist_dir = os.path.join(root, "dist", "POE-Ninja-three-language-v0.2.14")
zip_path = os.path.join(root, "dist", "POE-Ninja-three-language-v0.2.14.zip")

if os.path.exists(zip_path):
    os.remove(zip_path)

print(f"正在打包目录: {dist_dir}")
print(f"目标 ZIP 包: {zip_path}")

with zipfile.ZipFile(zip_path, "w", zipfile.ZIP_DEFLATED, compresslevel=9) as zipf:
    for foldername, subfolders, filenames in os.walk(dist_dir):
        for filename in filenames:
            filepath = os.path.join(foldername, filename)
            arcname = os.path.relpath(filepath, dist_dir)
            zipf.write(filepath, arcname)

size = os.path.getsize(zip_path)
print("\n====================================================")
print("最终可分发压缩包已成功生成！")
print(f"文件绝对路径: {zip_path}")
print(f"安装包最终体积: {size / 1024 / 1024:.2f} MB ({size / 1024:.1f} KB)")
print("====================================================")
